client/pov-recovery/src/lib.rs

```diff
@@ -302,12 +402,19 @@ where
 
 		match self.parachain_client.block_status(parent) {
 			Ok(BlockStatus::Unknown) => {
-				if self.active_candidate_recovery.is_being_recovered(&parent) {
+				// If the parent block is currently being recovered or is scheduled to be recovered,
+				// we want to wait for the parent.
+				let parent_scheduled_for_recovery =
+					self.candidates.get(&parent).map_or(false, |parent| parent.waiting_recovery);
+				if self.active_candidate_recovery.is_recovering(&parent) ||
+					parent_scheduled_for_recovery
```

Comment:
```suggestion
				let parent_being_recovered =
					self.candidates.get(&parent).map_or(false, |parent| parent.waiting_recovery);
				if parent_being_recovered
```

You just updated the comment above `waiting_recovery` saying that this is only `true` when you either waiting for recovery or it is being recovered.